app/bundles/LeadBundle/Entity/Lead.php

```diff
@@ -34,10 +34,7 @@ class Lead extends FormEntity implements CustomFieldEntityInterface, IdentifierF
      */
     private $availableSocialFields = [];
 
-    /**
-     * @var int
-     */
-    private $id;
+    private int $id;
```

Comment:
Entity properties must be nullable if they do not have default value or required by constructor. Imagine
```php
$lead = new Lead();
$lead->getId(); // <- will result in type error
```

Reply:
When properties and their getters are nullable but the db column isn't, a lot of type checking is added to make sure its not null whenever we need to use the value.

Now might be a good time to switch to uuids and move required fields into the constructor. WDYT?

Making them nullable in this PR.

Reply:
> When properties and their getters are nullable but the db column isn't, a lot of type checking is added to make sure its not null whenever we need to use the value.

This is true and that's what we use throughout the application to identify if it's a new entity or an existing one.

> Now might be a good time to switch to uuids and move required fields into the constructor. WDYT?

That would be nice but also massive refactoring. Right now we are almost 1 year in preparing M5 and we should stop doing unnecessary refactoring and focus on only what is necessary for M5 release.

BTW this PR:

https://github.com/mautic/mautic/pull/12262

is part of refactoring needed for Doctrine upgrade and it updates the entity properties. Give it a review, please.

Reply:
I mostly had the lead refactored so I went ahead and finished it, I'm going to move that to another branch and remove it from this one.

Because of https://github.com/mautic/mautic/pull/12262 I am going to do the minimum here.

I should have the review for that PR done today. So far its looking good. 

Ill do separate PRs based on that branch for stronger typing if there is time to do it.

Reply:
There is a rector rule that will be implemented to automate the process of moving fields into constructor, first we need to finish the doctrine upgrade.

Your review to this [PR](https://github.com/mautic/mautic/pull/12262) is highly appreciated, I will try to finish any pending feedback by tomorrow